homeassistant/components/husqvarna_automower/coordinator.py

```diff
@@ -77,17 +77,32 @@ async def _async_update_data(self) -> MowerDictionary:
             self.api.register_data_callback(self.handle_websocket_updates)
             self.ws_connected = True
         try:
-            data = await self.api.get_status()
+            self.data = await self.api.get_status()
```

Comment:
This is kinda weird writing to `self.data` here as the method itself ends with writing the result to `self.data`.

Reply:
Not entirely sure I understand what the problem is but if it's when we add/remove entities we could just create a new listener and use that to trigger the method after `async_refresh` e.g. when `self.data` has been written.

Reply:
Now I don't write `self.data` anymore in the update. The problem is, when adding entities, `self.data` has to be up to date. So a listener to start `self._async_add_remove_devices_and_entities(data)` on a data update would be good. But I'm not sure, where to place it, because the function needs the data from before updating `self.data`

Reply:
Hi @Thomas55555
I see you made some changes so is this ready for review now?